base/errorshow.jl

```diff
@@ -359,8 +359,10 @@ function show_method_candidates(io::IO, ex::MethodError, @nospecialize kwargs=()
         end
     end
 
-    for (func, arg_types_param) in funcs
-        for method in methods(func)
+    for (f, arg_types_param) in funcs
+        f2 = f isa UnionAll ? unwrap_unionall(unwrap_unionall(f).name.wrapper) : f
+        for method in methods(f2)
```

Comment:
`methods(unwrap_unionall(...))` is an invalid query—this must be called only on a properly wrapped object (e.g. the `.wrapper`)

Reply:
That was the original suggestion in https://github.com/JuliaLang/julia/issues/33539#issue-506110561, and it works obviously for `DataType`.
Using the `methods(... .name.wrapper)` does not deliver all the methods we are after.
How would I have to "properly wrap" the object?

Reply:
Ah, right. `methods` can't do this query properly (which is its own shortcoming). You'd need to bypass it and call into the `Base._methods_by_ftype` function directly with `f2 isa Type ? Type{<:f2} : typeof(f2)`.

Reply:
I don't see the point: If I `@enter methods(f2)` I end at
```
830      tt = signature_type(f, t)
831      return _methods_by_ftype(tt, lim, world)

1|julia> tt
Tuple{Type{Foo{T<:Integer,S,U}},Vararg{Any,N} where N}
```
That seems to be exactly the same, what you propose to call directly. The results are identical. Only difference is,
the `methods` throws an `ArgumentError`, if `f2 isa Core.Builtin`. So why shouldn't I simply call `methods(f2)`??